osu.Server.Spectator/LoggingHubFilter.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using osu.Server.Spectator.Hubs;
+
+namespace osu.Server.Spectator
+{
+    /// <summary>
+    /// An <see cref="IHubFilter"/> logging method invoke and error to the <see cref="ILoggableHub"/>.
+    /// </summary>
+    public class LoggingHubFilter : IHubFilter
+    {
+        public async ValueTask<object?> InvokeMethodAsync(HubInvocationContext invocationContext, Func<HubInvocationContext, ValueTask<object?>> next)
+        {
+            var loggableHub = invocationContext.Hub as ILoggableHub;
```

Comment:
would turn this into a guard to save on the repetitive checks, i.e.

```csharp
if (!(invocationContext.Hub is ILoggingHub loggingHub))
    return;
```

Reply:
I previously had `ILoggingHub` named as `IStatefulUserHub`, which made me concerned whether I want to do that. But now that I renamed it to `ILoggingHub`, would make sense to early-return now given that the entire hub filter is about logging.